### PR TITLE
Adds python project requests

### PR DESCRIPTION
**Pull request by catenacyber**
Need to move the fuzz target to the upstream repository

### CODE REVIEW

File: projects/requests/build.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash -eu
+# Copyright 2020 Google Inc.
```
Comment (jonathanmetzman) [inline projects/requests/build.sh:2]: Google LLC.

File: projects/requests/fuzz_server.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python3
+
+# Copyright 2020 Google LLC
```
Comment (jonathanmetzman) [inline projects/requests/fuzz_server.py:3]: 2021

File: projects/requests/build.sh (hunk 1 above)
Comment (jonathanmetzman) [reply projects/requests/build.sh:2]: 2021

**Comment by jonathanmetzman** [comment]
We'll accept this project if upstream approves.

**Comment by catenacyber** [comment]
cc @nateprewitt

**Comment by nateprewitt** [comment]
Thanks for the cc, @catenacyber. I took a look over the repo and wanted to quickly confirm my understanding. Are there any requirements or implications on Requests beyond receiving fuzz reports?

**Comment by catenacyber** [comment]
> Are there any requirements or implications on Requests beyond receiving fuzz reports?

For the reports (if any bug is found), would you like me to add more people ?

I do not think there are requirements, but there are some advices.
It is recommended to have the fuzz targets (ie fuzz_server.py) in the upstream repository.
Should I make a PR to add it in a tests/fuzz subdirectory ?

This is an initial integration.
After that, you can (if you want) improve it in multiple ways, cf https://google.github.io/oss-fuzz/advanced-topics/ideal-integration/
I would recommend the use of CIFuzz to get every Pull Request tested with fuzzing cf https://google.github.io/oss-fuzz/getting-started/continuous-integration/

File: projects/requests/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+FROM gcr.io/oss-fuzz-base/base-builder
+
+RUN git clone --depth 1 --branch master https://github.com/psf/requests.git
+
+WORKDIR $SRC/
```
Comment (inferno-chromium) [inline projects/requests/Dockerfile:21]: This needs to be $SRC/requests

File: projects/requests/build.sh [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# Build and install project (using current CFLAGS, CXXFLAGS).
+cd requests
```
Comment (inferno-chromium) [inline projects/requests/build.sh:19]: Not needed.

File: projects/requests/build.sh [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+pip3 install .
+
+# TODO move in upstream repo
```
Comment (inferno-chromium) [inline projects/requests/build.sh:22]: no need to add todo here.

**Review by inferno-chromium — CHANGES_REQUESTED** [review]

File: projects/requests/build.sh (hunk 4 above)
Comment (catenacyber) [reply projects/requests/build.sh]: ok

File: projects/requests/Dockerfile (hunk 3 above)
Comment (catenacyber) [reply projects/requests/Dockerfile]: ok

File: projects/requests/build.sh (hunk 5 above)
Comment (catenacyber) [reply projects/requests/build.sh]: ok

**Comment by catenacyber** [comment]
@nateprewitt do you have more questions ?
Are you interested in the continuous fuzzing of requests by oss-fuzz ?

**Comment by DavidKorczynski** [comment]
@catenacyber could you perhaps re-open this PR and put yourself as primary contact and me as an auto_ccs?

**Comment by catenacyber** [comment]
@DavidKorczynski like that ?

**Comment by DavidKorczynski** [comment]
> @DavidKorczynski like that ?

Yeah exactly, thanks! Could you also get the CI passing? See here what I had to do for taking an old Python PR into 2022: https://github.com/google/oss-fuzz/pull/4908/commits/f2bbb7b428acef3faf74e84486dd97a3d9e0bb78

**Comment by DavidKorczynski** [comment]
I think coverage is failing because of this just-merged PR: https://github.com/google/oss-fuzz/pull/7298

i.e. we can use `compile_python_fuzzer` to compile fuzzers now which will also fix coverage, but am not sure if the base image is updated yet.

**Review by DavidKorczynski — APPROVED** [review]
Thanks @catenacyber -- lgtm, I can assist with coverage in the coming days

**Comment by DavidKorczynski** [comment]
@oliverchang ping

**Review by DavidKorczynski — COMMENTED** [review]
I think the CI passes if you change:
```
  fuzzer_basename=$(basename -s .py $fuzzer)
  fuzzer_package=${fuzzer_basename}.pkg
  pyinstaller --distpath $OUT --onefile --name $fuzzer_package $fuzzer

  # Create execution wrapper.
  echo "#!/bin/sh
# LLVMFuzzerTestOneInput for fuzzer detection.
this_dir=\$(dirname \"\$0\")
ASAN_OPTIONS=\$ASAN_OPTIONS:symbolize=1:external_symbolizer_path=\$this_dir/llvm-symbolizer:detect_leaks=0 \
\$this_dir/$fuzzer_package \$@" > $OUT/$fuzzer_basename
  chmod u+x $OUT/$fuzzer_basename
```
to 
```
compile_python_fuzzer $fuzzer
```

**Review by oliverchang — APPROVED** [review]

**Comment by DavidKorczynski** [comment]
@inferno-chromium could you update your request for changes so the PR is approved?

**Pull request merged**